client/src/components/Navbar/Menus/Authenticated/Sidebar.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { makeStyles, Drawer, Button } from '@material-ui/core';
+import { GiHamburgerMenu } from 'react-icons/gi';
+import List from './List';
+
+const useStyles = makeStyles((theme) => ({
+  list: {
+    minWidth: 350,
+    width: 'auto',
+  },
+  burger: {
+    fontSize: 26,
+    [theme.breakpoints.down('md')]: {
+      fontSize: 22,
+    },
+  },
+}));
+
+const Sidebar = (): JSX.Element => {
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const classes = useStyles();
+
+  return (
+    <>
+      <Button onClick={() => setIsOpen(!isOpen)}>
```

Comment:
```suggestion
      <Button onClick={() => setIsOpen(prevOpen => !prevOpen)}>
```

For React, its best to not depend on its current state, rather than previous state because React does batch updates so if there are too many movements going on, you may run into bugs where the toggle is now different that what you expect

Reply:
It's a [best practice](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) to depend on previous state rather than current state as React does not guarantee that the state changes are applied immediately.